src/rt/aaA.d

```diff
@@ -152,6 +152,21 @@ private:
         deleted = 0;
         GC.free(obuckets.ptr); // safe to free b/c impossible to reference
     }
+
+    void clear() pure nothrow
+    {
+        foreach (ref b; buckets[firstUsed .. $])
+        {
+            if (b.filled)
+            {
+                b.hash = HASH_DELETED;
+                b.entry = null;
+            }
+        }
+        deleted = used;
+        firstUsed = cast(uint) dim;
+        resize(INIT_NUM_BUCKETS);
```

Comment:
> Well, that's definitely not a good reason to use clear. How would you even do that? You can't reserve any bucket data, and it would be silly to add a whole bunch of elements and then clear so you could reserve the buckets.

The use-case goes like this, parse a document and insert 1M entries, process the entries, clear the AA, continue w/ the next 1M inserts. Not having to resize the AA (and there are many resizes from 8 to 1M) would be a huge plus here.
Maybe you're right and it should be part of a separate reserve and capacity API, but killing the capacity on clear when we have a dedicated rehash for that would be remain weird.